test/parallel/test-net-write-arguments.js

```diff
@@ -3,15 +3,22 @@ const common = require('../common');
 const assert = require('assert');
 const net = require('net');
 
-const socket = net.Stream({ highWaterMark: 0 });
+{
+  const socket = net.Stream({ highWaterMark: 0 });
+
+  // Make sure that anything besides a buffer or a string throws.
+  socket.write(null, common.expectsError({
+    code: 'ERR_STREAM_NULL_VALUES',
+    name: 'TypeError',
+    message: 'May not write null values to stream'
+  }));
+  socket.on('error', common.expectsError({
+    code: 'ERR_STREAM_NULL_VALUES',
+    name: 'TypeError',
+    message: 'May not write null values to stream'
+  }));
+}
 
-// Make sure that anything besides a buffer or a string throws.
-assert.throws(() => socket.write(null),
-              {
-                code: 'ERR_STREAM_NULL_VALUES',
-                name: 'TypeError',
-                message: 'May not write null values to stream'
-              });
```

Comment:
Should this test be kept in place? Or is the error not thrown anymore in case there's no callback?
Similar question in test/parallel/test-net-socket-write-error.js

Reply:
The added code above covers this, also it doesn't `throw` it anymore it's always callback or `'error'`. Same thing in test/parallel/test-net-socket-write-error.js.